### ISSUE TITLE
histogramNumBins should produce the requested number of bins

### ISSUE THREAD

**Issue by bjartur** [mikeizbicki/histogram#1]
Histogram works, and is great for rapid plotting. The type checker is a helpful guide. There is, though, something left to be desired as to control over the exact number of bins in a histogram.

```
import Graphics.Histogram
import System.Random
import Data.Functor
import Data.Function
import Control.Monad

main=
      newStdGen <&> randomRs (0.04, 0.26) <&> take 28
                        <&> histogramNumBins 4
  >>= plot "unif.png" & void
```

![unif](https://user-images.githubusercontent.com/171073/79605045-be2df780-80de-11ea-958d-83c108bf29ba.png)

Requesting 4 bins resulted in 5 bins being drawn in 6 slots.

P.S. In the implementation of `histogramNumBins`, a function `firstDigit` is defined such that `firstdigit (0.3::Double)` is 2, and not 3, as the name implies. It returns the correct value for 0.5, but incorrect for 0.05 and 0.6, for example.

**Comment by mikeizbicki** [comment]
This is a great observation that should be fixed, but I'm not actively maintaining this library right now.

**Comment by bjartur** [comment]
Fair enough. As for firstDigit, multiple people have spent an inordinate amount of time compensating for approximations made by IEEE floating point numbers. `firstDigit = GHC.Float.floatToDigits <&> fst <&> head` would build on the work of GHC developers and the conference paper by R.G. Burger and R.K. Dybvig from ACM SIGPLAN 1996. GHC.Float.floatToDigits should probably be vendored in this library to preserve portability across compilers (including across versions and variants of GHC).
